Sources/Basic/OutputByteStream.swift

```diff
@@ -55,32 +58,108 @@ public class OutputByteStream: OutputStream {
         return buffer.count
     }
 
-    /// The contents of the output stream.
-    ///
-    /// This method implicitly flushes the stream.
-    public var bytes: ByteString {
-        flush()
-        return ByteString(self.buffer)
+    /// Currently available buffer size.
+    private var availableBufferSize: Int {
+        return max(buffer.capacity, OutputByteStream.bufferSize) - buffer.count
     }
-    
+
+    /// Clears the buffer maintaining current capacity.
+    private func clearBuffer() {
+        buffer.removeAll(keepingCapacity: true)
+    }
+
     // MARK: Data Output API
 
     public func flush() {
-        // Do nothing.
+        writeImpl(ArraySlice(buffer))
```

Comment:
I think this should use `writeIml` and then call a `flushImpl`, so the underlying stream can do whatever flush should actually do.